crates/rust-client/src/rpc/domain/accounts.rs

```diff
@@ -131,20 +132,41 @@ impl ProtoAccountHeader {
 // ------------------------------------------------------------------------------------------------
 
 impl ProtoAccountStateHeader {
-    pub fn into_domain(self, account_id: AccountId) -> Result<StateHeaders, RpcError> {
+    /// Attempts to convert the response into a valid [StateHeaders].
+    ///
+    /// The response only contains the account's code if it was updated compared to what we sent
+    /// as previously known (if any), so we need to provide it in case it was not sent.
```

Comment:
I'm not quite sure about the wording here. Feel free to disregard but I prefer something like:

```suggestion
    /// Attempts to convert the response into a valid [StateHeaders].
    ///
    /// The response includes the account's code only if it has been updated compared
    /// to the previously known version (if any). 
    /// Therefore, we must provide in case it was not sent.
```

Reply:
Thanks, changed!